src/components/tx-flow/common/TxNonce/index.tsx

```diff
@@ -136,21 +144,26 @@ const TxNonceForm = ({ nonce, recommendedNonce }: { nonce: number; recommendedNo
                     InputProps={{
                       ...params.InputProps,
                       name: field.name,
-                      endAdornment:
-                        !readOnly && recommendedNonce.toString() !== field.value ? (
-                          <InputAdornment position="end" className={css.adornment}>
-                            <Tooltip title="Reset to recommended nonce">
-                              <IconButton onClick={resetNonce} size="small" color="primary">
-                                <RotateLeftIcon fontSize="small" />
-                              </IconButton>
-                            </Tooltip>
-                          </InputAdornment>
-                        ) : null,
-                      readOnly,
+                      endAdornment: showRecommendedNonceButton ? (
+                        <InputAdornment position="end" className={css.adornment}>
+                          <Tooltip title="Reset to recommended nonce">
+                            <IconButton onClick={resetNonce} size="small" color="primary">
+                              <RotateLeftIcon fontSize="small" />
+                            </IconButton>
+                          </Tooltip>
+                        </InputAdornment>
+                      ) : null,
                     }}
-                    className={css.input}
+                    className={classNames([
+                      css.input,
+                      {
+                        [css.withAdornment]: showRecommendedNonceButton,
+                      },
+                    ])}
                     sx={{
-                      minWidth: `clamp(1ch, ${field.value.length}ch, 200px)`,
+                      minWidth: `clamp(1ch, calc(${field.value.toString().length}ch${
+                        showRecommendedNonceButton ? ' + 28px' : ' + 4px'
```

Comment:
I couldn't figure out a more elegant way to do handle the dynamic widths :/

Reply:
If we want to have the input as small as possible we need to take this into account so I think its fine.